server/data_access_layer/daos/name.dao.js

```diff
@@ -36,6 +37,112 @@ exports.getClientByID = async (clientIDList, db=database) => {
     });
 }
 
+
+exports.getClientByIDAndCountry = async (clientIDList, countryCode, db=database) => {
+    return new Promise(async (resolve, reject) => {
+        try{
+            const data = await db.query(
+                "SELECT DISTINCT N.NAME_ID, ISNULL(N.NAME_1,'')+ISNULL(' '+N.NAME_2,'')+ISNULL(' '+N.NAME_3,'') as NAME, C.COUNTRY_LABEL, C.COUNTRY_CODE \
+                FROM NAME N, NAME_CONNECTION NC, COUNTRY C \
+                WHERE NC.CONNECTION_NAME_ID in (?) \
+                AND NC.NAME_ID = N.NAME_ID \
+                AND N.LEGAL_COUNTRY_CODE = C.COUNTRY_CODE \
+                AND C.COUNTRY_LABEL = ? \
+                ORDER BY NAME",
+                {
+                    replacements: [clientIDList, countryCode],
```

Comment:
Here again, you can change the name of the `countryCode` variable to `countryLabel` if that's what you're actually passing